main.go

```diff
@@ -507,6 +510,16 @@ func deployGoogleEndpointsServiceIfRequired(ctx context.Context, params Params)
 	}
 }
 
+func failIfCreatingNewPublicService(ctx context.Context, params Params, templateData TemplateData, name, namespace string) {
+	if params.Kind == "deployment" && params.Visibility == "public" {
+		serviceType, err := foundation.GetCommandWithArgsOutput(ctx, "kubectl", []string{"get", "service", name, "-n", namespace, "-o=jsonpath={.spec.type}"})
```

Comment:
So if app is being migrated to a different cluster, user should change the visibility?

Reply:
As we decided to avoid public visibility, I think it makes sense not to allow creating public visibility service via Estafette. If there is no alternative and we agree to use such service, we can manually create it and this extension won't modify it.